docs/tutorial/creating_cli.md

```diff
@@ -0,0 +1,643 @@
+```
+
+That was rough, but congratulations on making it so far! Now we can move to the last part of our application.
+
+## Prompt
+The last part is the prompt parsing, the functionality that is going to parse the user command line arguments into instructions that Scala can understand and then we can pass to the `Cli`.
+
+We strongly suggest that you use a command parsing library, for example, [Decline](https://ben.kirw.in/decline/). 
+
+But again, for the sake of simplicity, well demonstrate how we achieved that without the need of an external dependency. 
+
+First thing is creating a parse function that will pattern match the `args` parameter of our entry point functions and, depending on the command, will call the respective function of the `Cli` class:
+
+```scala 3
+//src/main/shellfish/contacts/cli/Prompt.scala
+def parsePrompt(args: List[String]): CliCommand =
+  args match 
+    case "add" :: Nil                          => AddContact
+    case "remove" :: username :: Nil           => RemoveContact(username)
+    case "search" :: "id" :: username :: Nil   => SearchId(username)
+    case "search" :: "name" :: name :: Nil     => SearchName(name)
+    case "search" :: "email" :: email :: Nil   => SearchEmail(email)
+    case "search" :: "number" :: number :: Nil => SearchNumber(number)
+    case "list" :: _                           => ViewAll
+    case "update" :: username :: options =>
+      UpdateContact(username, parseUpdateFlags(options))
+    case Nil                                   => Help
+    case "--help" :: _ => Help
+    case "help" :: _   => Help
+    case _             => Help
+end parsePrompt
+```
+
+Here we're using pattern matching to match the command and the arguments, so we can create the corresponding `CliCommand`.
+
+Notice that we also created a `parseUpdateFlags` function, this does a similar thing, but it creates a list of flags that can change the updating behaviour:  
+
+```scala 3
+//src/main/shellfish/contacts/cli/Prompt.scala`
+private def parseUpdateFlags(options: List[String]): List[Flag] = 
+
+  @tailrec
+  def tailParse(remaining: List[String], acc: List[Flag]): List[Flag] =
+    remaining match 
+  
+      case Nil => acc
+  
+      case "--first-name" :: firstName :: tail =>
+        tailParse(tail, FirstNameFlag(firstName) :: acc)
+  
+      case "--last-name" :: lastName :: tail =>
+        tailParse(tail, LastNameFlag(lastName) :: acc)
+  
+      case "--phone-number" :: phoneNumber :: tail =>
+        tailParse(tail, PhoneNumberFlag(phoneNumber) :: acc)
+  
+      case "--email" :: email :: tail =>
+        tailParse(tail, EmailFlag(email) :: acc)
+  
+      case flag :: _ => List(UnknownFlag(flag))
+    
+  
+  tailParse(options, Nil)
+  
+end parseUpdateFlags
+```
+
+Because the update flags  are unordered, we use a recursive function to match each flag and its value, and then we create a list of `Flag` that will be passed to the `updateCommand` function like defined before.
```

Comment:
```suggestion
Because the update flags can be unordered, we case use a recursive function to match each flag and its value and then create a list of `Flag`.
```